includes/handler/class-create.php

```diff
@@ -34,14 +35,42 @@ public static function init() {
 	 * @param \Activitypub\Activity\Activity $activity_object Optional. The activity object. Default null.
 	 */
 	public static function handle_create( $activity, $user_id, $activity_object = null ) {
-		// Check if Activity is public or not.
-		if (
-			ACTIVITYPUB_CONTENT_VISIBILITY_PRIVATE === get_activity_visibility( $activity ) ||
-			! is_activity_reply( $activity )
-		) {
+		// Check for private and/or direct messages.
+		if ( ACTIVITYPUB_CONTENT_VISIBILITY_PRIVATE === get_activity_visibility( $activity ) ) {
+			$result = false;
+		} elseif ( is_activity_reply( $activity ) ) { // Check for replies.
+			$result = self::create_interaction( $activity, $user_id, $activity_object );
+		} else { // Handle non-interaction objects.
+			$result = self::create_post( $activity, $user_id, $activity_object );
+		}
+
+		if ( false === $result ) {
 			return;
 		}
 
+		$success = ( false !== $result && ! \is_wp_error( $result ) );
```

Comment:
```suggestion
		$success = ! \is_wp_error( $result );
```
`$result` is never false at this point.